src/os/ui/layer/vectorlayerui.js

```diff
@@ -360,23 +367,27 @@ os.ui.layer.VectorLayerUICtrl.prototype.loadPresets = function() {
       if (promise) {
         promise.then(function(presets) {
           if (presets && presets.length) {
-            var basicPreset = presets.find(function(p) {
-              return p.id === os.layer.preset.DEFAULT_PRESET_ID;
-            });
-            // tweak the colors to match the layer
-            if (basicPreset) {
-              os.layer.preset.updateDefault(layer, basicPreset);
-            }
-
             this['presets'] = presets;
 
             // the preset objects may change, so resolve the current selection by id
             var currentPreset = this['preset'] ? presets.find(function(p) {
               return p && p.id == this['preset'].id;
             }, this) : undefined;
 
-            // set the current selection, with priority as current > default > first
-            this['preset'] = currentPreset || basicPreset || presets[0];
+            // the preset might be saved in settings
+            var settingsPreset = null;
+            var settingsPresetId = os.layer.preset.getSavedPresetId(id);
+            if (settingsPresetId) {
+              settingsPreset = presets.find(function(p) {
+                return p && p.id == settingsPresetId;
+              }, this);
+            }
+
+            // set the current selection, with priority as current > settings > first
```

Comment:
This comment doesn't accurately represent the order. And do we no longer need to consider the default/basic preset?